app/code/Magento/CatalogGraphQl/Model/Resolver/Product/SpecialPrice.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model\Resolver\Product;
+
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Catalog\Pricing\Price\SpecialPrice as PricingSpecialPrice;
+
+/**
+ * Resolver for Special Price
+ */
+class SpecialPrice implements ResolverInterface
+{
+    /**
+     * @inheritdoc
+     */
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
+    {
+        /** @var \Magento\Catalog\Model\Product $product */
+        $product = $value['model'];
+        /** @var PricingSpecialPrice $specialPrice */
+        $specialPrice = $product->getPriceInfo()->getPrice(PricingSpecialPrice::PRICE_CODE);
```

Comment:
you already know how to get here customerGroup, so use it here to get the price you want, pass it by param